model/distro/distro.go

```diff
@@ -241,9 +241,12 @@ func (d *Distro) ShellBinary() string {
 }
 
 type HostAllocatorSettings struct {
-	Version                string `bson:"version" json:"version" mapstructure:"version"`
-	MinimumHosts           int    `bson:"minimum_hosts" json:"minimum_hosts" mapstructure:"minimum_hosts"`
-	MaximumHosts           int    `bson:"maximum_hosts" json:"maximum_hosts" mapstructure:"maximum_hosts"`
+	Version      string `bson:"version" json:"version" mapstructure:"version"`
+	MinimumHosts int    `bson:"minimum_hosts" json:"minimum_hosts" mapstructure:"minimum_hosts"`
+	MaximumHosts int    `bson:"maximum_hosts" json:"maximum_hosts" mapstructure:"maximum_hosts"`
+	// AutoTuneMaximumHosts determines if Evergreen is allowed to automatically
+	// tune the distro's maximum hosts.
+	AutoTuneMaximumHosts   bool   `bson:"auto_tune_maximum_hosts" json:"auto_tune_maximum_hosts" mapstructure:"auto_tune_maximum_hosts"`
```

Comment:
When we add this to the UI after testing, would we want this to be opt-out behavior? Having it opt-in for now makes sense to me for testing but I would think the flag we'd be adding to the UI would be the opposite, since most distros should probably do this. If we intend to set the flag automatically on distro creation to mimic opt-in that also works for me!